src/main/java/com/iota/iri/service/milestone/impl/LatestSolidMilestoneTrackerImpl.java

```diff
@@ -154,6 +173,39 @@ private void latestSolidMilestoneTrackerThread() {
         }
     }
 
+    /**
+     * Applies the given milestone to the ledger.<br />
+     * <br />
+     * If the application of the milestone fails, we start a repair routine which will revert the milestones preceding
+     * our current milestone and consequently try to reapply them in the next iteration of the {@link
+     * #checkForNewLatestSolidMilestones()} method (until the problem is solved).<br />
+     *
+     * @param milestone the milestone that shall be applied to the ledger state
+     * @throws Exception if anything unexpected goes wrong while applying the milestone to the ledger
+     */
+    private void applySolidMilestoneToLedger(MilestoneViewModel milestone) throws Exception {
+        if (!ledgerService.applyMilestoneToLedger(milestone)) {
+            revertPrecedingMilestones(milestone);
+        } else {
+            resetRepairBackoffCounterIfProblemSolved(milestone);
+        }
+    }
+
+    /**
+     * Resets the internal variables that are used to keep track of the repair process.<br />
+     * <br />
+     * It gets called whenever we advance to a milestone that has a higher milestone index than the milestone that
+     * initially caused the repair routine to kick in (see {@link #revertPrecedingMilestones(MilestoneViewModel)}.<br />
+     *
+     * @param processedMilestone the milestone that currently gets processed
+     */
+    private void resetRepairBackoffCounterIfProblemSolved(MilestoneViewModel processedMilestone) {
```

Comment:
this name is a bit misleading, it's also the common case, where there was no problem to begin with.

Reply:
changed it and introduces some additional utility methods for better readability